### ISSUE TITLE
Support for Python 3.8 to be dropped

### ISSUE THREAD

**Issue by billybednar** [tprelog/iocage-homeassistant#47]
This plugin is using Python 3.8 which will no longer be supported in January. Per the [2021.11 release notes](https://www.home-assistant.io/blog/2021/11/03/release-202111/):
> Python 3.8 has been deprecated because Python 3.10 has been released. Support will be dropped in Home Assistant 2022.1.
>
>Support for Python 3.10 hasn’t completed yet, however we aim to complete that before the 3.8 deprecation period has ended. It is advised to upgrade to Python 3.9 at this point.

**Comment by tprelog** [comment]
Thanks for opening an issue @billybednar

I expect to have this plugin updated by the next (2021.12) release. But you don't have to wait!

Use these commands to:
- install the requirements for Python 3.9
- set the version for Home Assistant
- rebuild the Home Assistant virtualenv

```bash
iocage console $JAIL_NAME
# press 0 to exit menu
pkg install -y python39 py39-sqlite3
sysrc homeassistant_python=/usr/local/bin/python3.9
service homeassistant reinstall homeassistant
```

**Comment by adorobis** [comment]
Thanks, it worked (almost) ok:
```
 Python 3.9.7
 Home Assistant 2021.11.1
```
The only missing bit was `pymysql` package - I've installed it manually in the venv and everything seems to work OK:
```
2021-11-06 17:37:46 ERROR (Recorder) [homeassistant.components.recorder] Error during connection setup to mysql+pymysql://xx:yy@10.144.1.141/db_ha?charset=utf8mb4: No module named 'pymysql'
```

**Comment by tprelog** [comment]
I don't have that error with a clean install but I can confirm `pymysql` was _not_ installed by default

> The only missing bit was `pymysql` package

Are you using the default database? Or maybe that's a requirement for one of the integrations your using?

> I've installed it manually in the venv

FYI, you can install more python packages in the homeassistant virtualenv like this

```
service homeassistant install pymysql
```

If you know about extra python packages ahead of time, you can include during the reinstall like this. Note, for this to work `homeassistant` has to be the first in the list

```
service homeassistant reinstall homeassistant pymysql
```

**Comment by adorobis** [comment]
> Are you using the default database?

No, it is the MariaDB for recorder component. I was a bit surprised that pymysql would not install automatically - any other dependency for standard and custom components installed nicely.

> FYI, you can install more python packages in the homeassistant virtualenv like this

Wow, this is much easier than going into the venv. Thanks!

**Comment by tprelog** [comment]
I'm going to merge a quick update to plugin version 7 - This will at least start using Python 3.9 as default for all clean installs - It should also _switch existing services over to Python 3.9 but only if you reinstalled them._ This will also ensure `python39` and `py39-sqlite3` are installed

Running a Plugin UPDATE will basically ensure all the requirements are met and set the default python version to 3.9

I think I'm gonna leave it up the end user to reinstall the services. 

So after a user updates to version 7 - all of their installed services will continue to run a virtualenv with python 3.8.
After a user updates to version 7 - switch a virtualenv to python 3.9 by reinstalling the service
